python/samples/getting_started_with_processes/step01_processes.py

```diff
@@ -0,0 +1,210 @@
+
+
+class KernelProcessStep(Generic[TState]):
+    def __init__(self):
+        self._state: TState | None = None
+
+    async def activate(self, state: "KernelProcessStepState[TState]"):
+        """Activates the step and sets the state."""
+        pass
+
+    def create_default_state(self) -> TState:
+        """Creates the default state object."""
+        return None
+
+    async def on_activate(self):
+        """To be overridden by subclasses if needed."""
+        pass
+
+
+class ScriptedUserInputStep(KernelProcessStep[UserInputState]):
+    class Functions:
+        GetUserInput = "get_user_input"
+
+    def create_default_state(self) -> "UserInputState":
+        """Creates the default UserInputState."""
+        return UserInputState()
+
+    def populate_user_inputs(self):
+        """Method to be overridden by the user to populate with custom user messages."""
+        pass
+
+    async def on_activate(self):
+        """This is called during the activation of the process step."""
+        self.populate_user_inputs()
+
+    async def activate(self, state: "KernelProcessStepState[TState]"):
+        """Activates the step and sets the state."""
+        state.state = state.state or self.create_default_state()
+        self._state = state.state
+        self.populate_user_inputs()
+        pass
+
+    @kernel_function(name=Functions.GetUserInput)
+    async def get_user_input(self, context: KernelProcessStepContext):
+        """Gets the user input."""
+        if not self._state:
+            raise ValueError("State has not been initialized")
+
+        user_message = self._state.user_inputs[self._state.current_input_index]
+
+        print(f"USER: {user_message}")
+
+        if "exit" in user_message:
+            await context.emit_event(KernelProcessEvent(id=ChatBotEvents.Exit, data=None))
+            return
+
+        self._state.current_input_index += 1
+
+        # Emit the user input event
+        await context.emit_event({"id": CommonEvents.UserInputReceived, "data": user_message})
+
+
+class ChatUserInputStep(ScriptedUserInputStep):
+    def populate_user_inputs(self):
+        """Override the method to populate user inputs for the chat step."""
+        if self._state is not None:
+            self._state.user_inputs.append("Hello")
+            self._state.user_inputs.append("How tall is the tallest mountain?")
+            self._state.user_inputs.append("How low is the lowest valley?")
+            self._state.user_inputs.append("How wide is the widest river?")
+            self._state.user_inputs.append("exit")
+
+
+class IntroStep(KernelProcessStep):
+    @kernel_function
+    async def print_intro_message(self):
+        print("Welcome to Processes in Semantic Kernel.\n")
+
+
+class ChatBotState(KernelBaseModel):
+    """The state object for ChatBotResponseStep."""
+
+    chat_messages: list = []
+
+
+SERVICE_ID = "default"
+
+
+class ChatBotResponseStep(KernelProcessStep[ChatBotState]):
+    class Functions:
```

Comment:
Question: why is it necessary to create a class here for `GetChatResponse`? I see this pattern in other places too.

Reply:
The class is the chunk of code that will eventually get instantiated during the process. We first take the type of the step, and then later, once the process is built and ready to run, it will initialize each step where it created the concrete object. 

Does this help answer your question?

Reply:
Clean up to use class attributes. Thanks for the callout.